packages/lesswrong/server/vulcan-lib/apollo-server/authentication.tsx

```diff
@@ -165,16 +173,20 @@ const ResetPasswordToken = new EmailTokenType({
     const validatePasswordResponse = validatePassword(password)
     if (!validatePasswordResponse.validPassword) throw Error(validatePasswordResponse.reason)
 
-    await updateMutator({ 
-      collection: Users,
-      documentId: user._id,
-      set: {
-        'services.password.bcrypt': await createPasswordHash(password),
-        'services.resume.loginTokens': []
-      } as any,
-      unset: {},
-      validate: false,
-    });
+    if (Users.isPostgres()) {
+      await new UsersRepo().resetPassword(user._id, await createPasswordHash(password));
+    } else {
+      await updateMutator({ 
+        collection: Users,
+        documentId: user._id,
+        set: {
+          'services.password.bcrypt': await createPasswordHash(password),
+          'services.resume.loginTokens': []
+        } as any,
+        unset: {},
+        validate: false,
+      });  
```

Comment:
Are these also slow and that's why you're changing them?

Reply:
These are broken because of an issue with how the query builder translates $set operations on nested fields, where the value we set is an array.  Copying over from slack:

The following translated query has an unfortunate bug:
  sql: `UPDATE "Users" SET "services" = JSONB_SET( "services" , '{resume, loginTokens}' ::TEXT[], $1 , TRUE) WHERE _id IN ( SELECT "_id" FROM "Users" WHERE "_id" =  $2 LIMIT $3 FOR UPDATE )  RETURNING "_id"`,
  args: [ [], 'REDACTED_USER_ID', 1 ]
This comes from await Users.rawUpdateOne({_id: user._id}, {$set: {"services.resume.loginTokens": []}}); , which is triggered in a callback when a user is banned.
The result is that services.resume.loginTokens ends up getting set to an empty object ({}) rather than an empty array ([]).  Adding a ::JSONB type cast to the parameterized argument ($1::JSONB) fixes it.